File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int N = height.length;
        int maxWater = Integer.MIN_VALUE;
        int left = 0, right = N - 1;

        int currWidth, currHeight;

        while (left < right) {
            currWidth = right - left;
            currHeight = Math.min(height[left], height[right]);

            maxWater = Math.max(maxWater, currWidth*currHeight);
            if (height[left] < height[right]) {
                left++;
            } else {
                right--;
            }
        }
        return maxWater;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height) - 1
        maxArea = 0
        
        while (left <= right):
            width = right - left
            length = min(height[left], height[right])
            maxArea = max(maxArea, width * length)
            if height[left] <= height[right]:
                left = left + 1
            else:
                right = right - 1
        
        return maxArea
            
        